scripts/create_gw5000_database.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+import re
+from mendeleev import element
+from collections import Counter
+from tqdm import tqdm
+from dateutil import tz
+from datetime import datetime
+
+def convert_smiles_to_molecular_formula(inchi: str) -> str:
+    ids_slash = [m.start() for m in re.finditer('/', inchi)]
+    start_idx = ids_slash[0] + 1
+    end_idx = ids_slash[1]
+
+    molecular_formula = inchi[start_idx: end_idx]
+    return  molecular_formula
+
+
+def calc_custom_molecular_formula(xyz):
+    atoms = []
+    for line in xyz.split('\n'):
+        str_regex = '[a-zA-Z]+'
+        atom_name = re.findall(str_regex, line)
+
+        atoms.extend(atom_name)
+
+    counter = Counter(atoms)
+
+    atom_names = list(counter.keys())
+    n_atoms = list(counter.values())
+
+    molecular_formula_info = []
+    for atom, n in zip(atom_names, n_atoms):
+        molecular_formula_info.append(atom)
+        molecular_formula_info.append(str(n))
+
+    custom_molecular_formula = ''.join(molecular_formula_info)
+    return custom_molecular_formula
+
+
+def calculate_n_electrons_in_mol(xyz):
+    molecular_formula = calc_custom_molecular_formula(xyz)
+    # corner case: C20H19AsCl2P
+
+    n_electrons = 0
+    num_regex = "[0-9]+"
+    pos_of_numbers = [
+        sub_str.start() for sub_str in re.finditer(num_regex, molecular_formula)
+    ]
+    str_regex = '[a-zA-Z]+'
+    pos_of_atomic_symbols = [
+        sub_str.start() for sub_str in re.finditer(str_regex, molecular_formula)
+    ]
+
+    for pos_atom_symbol, pos_n_atoms in zip(pos_of_atomic_symbols, pos_of_numbers):
+        atom_symbol = molecular_formula[pos_atom_symbol:pos_n_atoms]
+        n_atoms = int(molecular_formula[pos_n_atoms])
+        atom = element(atom_symbol)
+        atom_number = atom.atomic_number
+        n_electrons += atom_number*n_atoms
+
+    return n_electrons
+
+def remove_xyz_header(xyz):
+    end_idx_header = [m.start() for m in re.finditer('\n', xyz)][2]
+    xyz_header_removed = xyz[end_idx_header+1:]
+    return xyz_header_removed
+
+
+def write_geom_database(mol_info_filepath, out_geom_database_filepath):
+    s_time = datetime.now(tz=tz.gettz('Europe/Stockholm'))
```

Comment:
Shall we remove the milliseconds? It doesn't seem to be necessary.

Reply:
Removed.